### PR TITLE
add self-destruct contract unit test

### LINKED ISSUE #2618: Unit test for selfdestruct
We need a unit test to make sure selfdestruct (https://solidity-by-example.org/hacks/self-destruct/) works well with our chain/EVM.

### PR DESCRIPTION
**Pull request by Liuhaai**
fix #2618

### CODE REVIEW

File: action/protocol/execution/protocol_test.go [hunk 1]
```diff
@@ -868,6 +868,10 @@ func TestProtocol_Handle(t *testing.T) {
 	t.Run("infiniteloop-bering", func(t *testing.T) {
 		NewSmartContractTest(t, "testdata/infiniteloop-bering.json")
 	})
+	// self-destruct
+	t.Run("self-destruct", func(t *testing.T) {
+		NewSmartContractTest(t, "testdata/self-destruct.json")
+	})
```
Comment (dustinxie) [inline action/protocol/execution/protocol_test.go:874]: also replicate the test in `testdata-istanbul` folder, add to `TestIstanbulEVM`

File: action/protocol/execution/testdata/self-destruct.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        "rawByteCode": "2e64cec1",
+        "rawAmount": "0",
+        "rawGasLimit": 1000000,
+        "rawGasPrice": "0",
+        "rawExpectedGasConsumed": 10400,
+        "hasReturnValue": true,
+        "rawReturnValue": "",
+        "comment": "return nil because contract is destoried"
+    }]
+}
```
Comment (dustinxie) [inline action/protocol/execution/testdata/self-destruct.json:57]: remember to `make fmt` and `make lint` before submitting PR

**Review by raullenchai — APPROVED** [review]

File: action/protocol/execution/testdata-istanbul/self-destruct.sol [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ function store(uint256 num) public {
+  number = num;
+ }
+ function retrieve() public view returns (uint256){
+  return number;
+ }
+ 
+ function close() public { 
+  selfdestruct(owner); 
+ }
```
Comment (CoderZhi) [inline action/protocol/execution/testdata-istanbul/self-destruct.sol:17]: indents

File: action/protocol/execution/testdata/self-destruct.sol [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+ function store(uint256 num) public {
+  number = num;
+ }
+ function retrieve() public view returns (uint256){
+  return number;
+ }
+ 
+ function close() public { 
+  selfdestruct(owner); 
+ }
```
Comment (CoderZhi) [inline action/protocol/execution/testdata/self-destruct.sol:17]: indents

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**